numpy/core/src/umath/ufunc_object.c

```diff
@@ -5219,7 +5219,11 @@ PyUFunc_RegisterLoopForDescr(PyUFuncObject *ufunc,
             if (cmp == 0 && current != NULL && current->arg_dtypes == NULL) {
                 current->arg_dtypes = PyArray_malloc(ufunc->nargs *
                     sizeof(PyArray_Descr*));
-                if (arg_dtypes != NULL) {
+                if (current->arg_dtypes == NULL) {
+                    PyErr_NoMemory();
+                    result = -1;
```

Comment:
I guess the assignment `current->nargs = ufunc->nargs;` is harmless, but it might be better to future-proof this by adding `goto done;` when an error condition is reached.